src/components/MapView/index.tsx

```diff
@@ -331,38 +334,57 @@ function MapView({ classes }: MapViewProps) {
-          if (featureInfoLayers.length === 0) {
-            return;
-          }
-
-          const dateFromRef = moment(
-            selectedDateRef.current?.props.selected as Date,
-          ).format('YYYY-MM-DD');
-
-          const params = getFeatureInfoParams(map, evt, dateFromRef);
-          dispatch(setWMSGetFeatureInfoLoading(true));
-          makeFeatureInfoRequest(featureInfoLayers, params).then(
-            (result: { [name: string]: string } | null) => {
-              if (result) {
-                Object.keys(result).forEach((k: string) => {
-                  dispatch(
-                    addPopupData({
-                      [k]: {
-                        data: result[k],
-                        coordinates: evt.lngLat,
-                      },
-                    }),
-                  );
-                });
+        onClick={
+          // this function will only work when boundary data loads.
+          // due to how the library works, we can only set this function once,
+          // so we should set it when boundary data is present
+          boundaryLayerData
+            ? (map: Map, evt: any) => {
+                dispatch(hidePopup());
+                // Hide the alert popup if we click outside the target country (outside boundary bbox)
+                if (
+                  boundaryLayerData.data.features.every(
+                    feature =>
+                      !inside(
+                        [evt.lngLat.lng, evt.lngLat.lat],
+                        feature as Feature<MultiPolygon>,
+                      ),
+                  )
+                ) {
+                  setIsAlertFormOpen(false);
+                }
+                // Get layers that have getFeatureInfo option.
+                const featureInfoLayers = getActiveFeatureInfoLayers(map);
+                if (featureInfoLayers.length === 0) {
+                  return;
+                }
+
+                const dateFromRef = moment(
+                  selectedDateRef.current?.props.selected as Date,
+                ).format('YYYY-MM-DD');
+
+                const params = getFeatureInfoParams(map, evt, dateFromRef);
+                dispatch(setWMSGetFeatureInfoLoading(true));
+                makeFeatureInfoRequest(featureInfoLayers, params).then(
+                  (result: { [name: string]: string } | null) => {
+                    if (result) {
+                      Object.keys(result).forEach((k: string) => {
+                        dispatch(
+                          addPopupData({
+                            [k]: {
+                              data: result[k],
+                              coordinates: evt.lngLat,
+                            },
+                          }),
+                        );
+                      });
+                    }
+                    dispatch(setWMSGetFeatureInfoLoading(false));
+                  },
+                );
```

Comment:
let's extract this into a separate function to keep it clean

Reply:
Yup definitely good since this is getting quite complex